package.json

```diff
@@ -0,0 +1,47 @@
+{
+  "name": "root",
+  "private": true,
+  "devDependencies": {
+    "@rollup/plugin-commonjs": "^21.0.0",
+    "@rollup/plugin-node-resolve": "^13.0.5",
+    "@types/mocha": "^9.1.0",
+    "@types/node": "^17.0.22",
+    "@types/node-fetch": "^2.6.1",
+    "@types/sinon": "^10.0.4",
+    "@typescript-eslint/eslint-plugin": "^5.16.0",
+    "@typescript-eslint/parser": "^5.16.0",
+    "eslint": "^8.11.0",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-simple-import-sort": "^7.0.0",
+    "prettier": "^2.6.0",
+    "rollup": "2.58.0",
+    "rollup-plugin-terser": "^7.0.2",
+    "sinon": "^13.0.1",
+    "typescript": "^4.6.2",
+    "lerna": "^4.0.0",
+    "@types/chai": "^4.3.0",
+    "chai": "4.3.4",
+    "mocha": "^6.2.3",
+    "esm": "^3.2.25",
+    "karma": "^6.3.4",
+    "karma-chai": "^0.1.0",
+    "karma-chrome-launcher": "^3.1.0",
+    "karma-firefox-launcher": "^2.1.1",
+    "karma-mocha": "^2.0.1",
+    "karma-typescript": "^5.5.2"
+  },
+  "workspaces": [
+    "packages/test/",
+    "packages/abstractions",
+    "packages/http/*",
+    "packages/serialization/json/",
+    "packages/serialization/text/"
+  ],
+  "scripts": {
+      "build": "npm run build:cjs && npm run build:es",
+      "build:cjs": "tsc -b packages/test/tsconfig.cjs.json",
```

Comment:
also it seems that the default build scripts that'll be used for publishing relies on the test tsconfig file. this probably needs to be updated.

Reply:
The build scripts build all the packages as they are linked to the test tsconfig file using project references. This ensures the the code is building with the latest changes and in the right order, that is, example abstractions is built before http.